include/envoy/stream_info/stream_info.h

```diff
@@ -479,8 +479,7 @@ class StreamInfo {
    * filters (append only). Both object types can be consumed by multiple filters.
    * @return the filter state associated with this request.
    */
-  virtual const FilterStateSharedPtr& filterState() PURE;
-  virtual const FilterState& filterState() const PURE;
+  virtual const FilterStateSharedPtr& filterState() const PURE;
```

Comment:
This change is a bit unfortunate; it's to make it possible to get the parent `FilterStateSharedPtr` from a const handle. Open for suggestions here for how to do this better

Reply:
I'd rather have a const-cast somewhere; this looks on the surface like it is const-safe, but the returned ptr is non-const, even though the method is const.

Reply:
Actually, I think I would make `onPoolReady()` pass a non-const StreamInfo. I think it's reasonable for that function to have non-const access to the StreamInfo.